app.js

```diff
@@ -1,8 +1,127 @@
 // import functions and grab DOM elements
+import { renderPastPoll } from './render-utils.js';
+
+const newPollFormEl = document.getElementById('new-poll-form');
+
+const currentPollQuestion = document.getElementById('current-poll-question');
+const optionAEl = document.getElementById('option-a');
+const optionASubtractButton = document.getElementById('option-a-subtract');
+const optionAVoteCountEl = document.getElementById('option-a-vote-count');
+const optionAAddButton = document.getElementById('option-a-add');
+const optionBEl = document.getElementById('option-b');
+const optionBSubtractButton = document.getElementById('option-b-subtract');
+const optionBVoteCountEl = document.getElementById('option-b-vote-count');
+const optionBAddButton = document.getElementById('option-b-add');
+
+const closePollButton = document.getElementById('close-poll-button');
+
+const pastPollsDivEl = document.getElementById('past-polls-div');
 
 // let state
+let currentPoll = {
+    question: '',
+    optionA: '',
+    optionB: '',
+    votesA: 0,
+    votesB: 0
+};
+
+const pastPollsArray = [];
 
 // set event listeners 
-  // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+optionASubtractButton.addEventListener('click', () => {
+    currentPoll.votesA--;
+    optionAVoteCountEl.textContent = currentPoll.votesA;
+});
+
+optionAAddButton.addEventListener('click', () => {
+    currentPoll.votesA++;
+    optionAVoteCountEl.textContent = currentPoll.votesA;
+});
+
+optionBSubtractButton.addEventListener('click', () => {
+    currentPoll.votesB--;
+    optionBVoteCountEl.textContent = currentPoll.votesB;
+});
+
+optionBAddButton.addEventListener('click', () => {
+    currentPoll.votesB++;
+    optionBVoteCountEl.textContent = currentPoll.votesB;
+});
+
+newPollFormEl.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const formData = new FormData(newPollFormEl);
+
+    // get user input
+    const userQuestion = formData.get('user-question-input');
+    const userOptionA = formData.get('option-a-input');
+    const userOptionB = formData.get('option-b-input');
+
+    // use user input to update state
+    currentPoll.question = userQuestion;
+    currentPoll.optionA = userOptionA;
+    currentPoll.optionB = userOptionB;
+
+    newPollFormEl.reset();
+
+    // update DOM to reflect the new state
+    displayCurrentPoll();
+});
+
+closePollButton.addEventListener('click', () => {
+  
+    const poll = renderPoll();
+  
+    pastPollsArray.push(poll);
```

Comment:
`renderPoll()` looks like it is just returning an identical object to your `currentPoll` object in state, so you could do a little less work and just push `currentPoll` into your `pastPollsArray`

Reply:
Awesome, this is much simpler. Thanks Vic!